### DISCUSSION TITLE
v2.5.0-rc1

### DISCUSSION THREAD

**Discussion by mashb1t** [lllyasviel/Fooocus#3282] [General]
## What's Changed

* feat: update python dependencies and add segment_anything
* sync enhance feature and code refactoring from [mashb1t/Fooocus](https://github.com/mashb1t/Fooocus/tree/main)
* feat: add enhance feature, which offers easy image refinement steps (similar to [adetailer](https://github.com/Bing-su/adetailer), but based on dynamic image detection instead of specific mask detection models). See https://github.com/lllyasviel/Fooocus/discussions/3281.
* feat: improve GroundingDINO and SAM image masking
* refactor: rewrite async worker code, make code much more reusable to allow iterations and improve reusability
* refactor: move checkboxes Enable Mask Upload and Invert Mask When Generating from Developer Debug Mode to Inpaint Or Outpaint
* refactor: rename checkbox Enable Mask Upload to Enable Advanced Masking Features
* fix: get upscale model filepath by calling downloading_upscale_model() to ensure the model exists
* i18n: rename tab titles and translations from singular to plural
* i18n: rename document to documentation
* feat: update default models to latest versions
  * animaPencilXL_v400 => animaPencilXL_v500 (https://github.com/lllyasviel/Fooocus/pull/2943)
  * DreamShaperXL_Turbo_dpmppSdeKarras => DreamShaperXL_Turbo_v2_1
  * SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4 => SDXL_FILM_PHOTOGRAPHY_STYLE_V1
*  feat: add preset for pony_v6 (using ponyDiffusionV6XL, discussion in https://github.com/lllyasviel/Fooocus/discussions/3217)
* feat: add style `Fooocus Pony` (discussion in https://github.com/lllyasviel/Fooocus/discussions/3217)
* feat: remove `by wlop` from style `Fooocus Masterpiece` as it has been causing unintended watermarks
* feat: add restart sampler ([paper](https://arxiv.org/abs/2306.14878))
* feat: add config option for default_inpaint_engine_version, sets inpaint engine for pony_v6 and playground_v2.5 to None for improved results (incompatible with inpaint engine, discussion in https://github.com/lllyasviel/Fooocus/discussions/3217)
* feat: add image editor functionality to mask upload (same as for inpaint, now correctly resizes and allows more detailed mask creation, discussion in https://github.com/mashb1t/Fooocus/discussions/44)
* feat: add persistent model cache for metadata. Use `--rebuild-hash-cache` to manually rebuild the cache for all non-cached hashes (optional, cache will otherwise be lazy-loaded. Remove `--rebuild-hash-cache` after executing once)
* refactor: rename `--enable-describe-uov-image` to `--enable-auto-describe-image` to better reflect its purpose (now also works for enhance image upload)
* feat: adjust playground_v2.5 preset by @mashb1t in https://github.com/lllyasviel/Fooocus/pull/3136
* fix: correctly identify and remove performance LoRA by @mashb1t in https://github.com/lllyasviel/Fooocus/pull/3150
* apply performance from metadata by @mashb1t in https://github.com/lllyasviel/Fooocus/pull/3153
* hotfix: add missing method in performance enum by @mashb1t in https://github.com/lllyasviel/Fooocus/pull/3154
* feat: add vae to possible preset keys by @mashb1t in https://github.com/lllyasviel/Fooocus/pull/3177
* feat: add restart sampler by @licyk in https://github.com/lllyasviel/Fooocus/pull/3219
* build(deps): bump docker/build-push-action from 5 to 6 by @dependabot in https://github.com/lllyasviel/Fooocus/pull/3223

## New Contributors
* @licyk made their first contribution in https://github.com/lllyasviel/Fooocus/pull/3219
* @dependabot made their first contribution in https://github.com/lllyasviel/Fooocus/pull/3223

**Full Changelog**: https://github.com/lllyasviel/Fooocus/compare/v2.4.3...v2.5.0-rc1

<hr /><em>This discussion was created from the release <a href='https://github.com/lllyasviel/Fooocus/releases/tag/v2.5.0-rc1'>v2.5.0-rc1</a>.</em>

**Comment by mrhan1993** [comment]
A huge update

If anyone encounters the following error during installation, it is because there is an issue with the `groundingdino-py` package. If you know how to proceed, download its tar package, modify line 41 in setup.py to `with open('readme.md', 'r+', encoding='utf-8') as readme:` and insider the folder, then run `pip install -e .` to install.

is work for me

```text
Defaulting to user installation because normal site-packages is not writeable
Looking in indexes: https://mirror.nju.edu.cn/pypi/web/simple
Collecting groundingdino-py==0.4.0
  Downloading https://mirror.nju.edu.cn/pypi/web/packages/da/26/af35089119098ecd9d2e174b9cfbeb0c1246d65449c718b4e57328a08c1c/groundingdino-py-0.4.0.tar.gz (82 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 82.3/82.3 kB 4.8 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [7 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\LiveR\AppData\Local\Temp\pip-install-x2mhlg21\groundingdino-py_d95a0eee1ce8466981beb059ff6a26bf\setup.py", line 41, in <module>
          readme = readme_file.read()
                   ^^^^^^^^^^^^^^^^^^
      UnicodeDecodeError: 'gbk' codec can't decode byte 0xa4 in position 2878: illegal multibyte sequence
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

**Reply by mashb1t** [reply]
I'll add a new zip file for the final release including all packages.

**Reply by fatinghenji** [reply]
Looking forward to the new release!

**Reply by mashb1t** [reply]
Workaround is to change the system language for the installation only, see https://github.com/IDEA-Research/GroundingDINO/issues/206

**Reply by aillnicy** [reply]
no offence but, i mean if IDK how to do what to do after reading this....?

**Reply by mashb1t** [reply]
@aillnicy if you're not familiar with the python ecosystem or python in general you can remove and reinstall Fooocus using the zip file https://github.com/lllyasviel/Fooocus/releases/download/v2.5.0/Fooocus_win64_2-5-0.7z
